### PR TITLE
Vivek/sqlmethod

### PR DESCRIPTION
**Pull request by vivekmenezes**

### CODE REVIEW

File: sql/wire/wire.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Vivek Menezes (vivek@cockroachlabs.com)
+
+package sqlwire
+
+//go:generate stringer -type=Method
```
Comment (petermattis) [inline sql/wire/wire.go]: You should put the `go:generate stringer` statement next to the type that is having a string method defined on it (i.e. in `method.go`).

File: sql/wire/http.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// for names of contributors.
+//
+// Author: Vivek Menezes (vivek@cockroachlabs.com)
+
+package sqlwire
+
+const (
+	// SQLDBEndpoint is the URL path prefix which accepts incoming
+	// HTTP requests for the SQL API.
+	SQLDBEndpoint = "/sql/db/"
```
Comment (petermattis) [inline sql/wire/http.go]: We're already in the `sqlwire` package: `s/SQLDBEndpoint/Endpoint/g`. And `s,/sql/db/,/sql/,g`.

File: sql/wire/http.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	// StatusTooManyRequests indicates client should retry due to
+	// server having too many requests.
+	// Defined here because it's still not a part of the HTTP standard.
+	StatusTooManyRequests = 429
```
Comment (petermattis) [inline sql/wire/http.go]: This was in `client/http_sender.go`, but I'm not sure if the server was ever sending back this status code. Perhaps drop from this PR until we actually have code that uses it.

**Comment by bdarnell** [comment]
This is a lot of files to define a few constants. Unless we anticipate a lot of http-related or method-related code to be added I'd just put the contents of http.go and method.go into wire.go.

LGTM

File: sql/wire/http.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Vivek Menezes (vivek@cockroachlabs.com)
+
+package sqlwire
```
Comment (bdarnell) [inline sql/wire/http.go]: My proposal was that the directory be named `sql/sqlwire`; I'd prefer to have the package and directory names match.

Comment (vivekmenezes) [inline sql/wire/http.go]: ah ha, got it. Let me send you another PR for that

File: sql/wire/http.go (hunk 3 above)
Comment (vivekmenezes) [inline sql/wire/http.go]: done

File: sql/wire/http.go (hunk 2 above)
Comment (vivekmenezes) [inline sql/wire/http.go]: done

File: sql/wire/wire.go (hunk 1 above)
Comment (vivekmenezes) [inline sql/wire/wire.go]: done
